### PR TITLE
PWN-4396 - Implement token list with new swipe layout

### PR DESCRIPTION
**Pull request by davran1001**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-4396

Redesign main token adapter item, add new swipe layout(Icon of hidden token was not changed, cause i didn't find second state)
Screen Shot 2022-08-08 at 02 22 52
Screen Shot 2022-08-08 at 02 23 09
Screen Shot 2022-08-08 at 02 23 32

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/home/ui/main/adapter/TitleViewHolder.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package org.p2p.wallet.home.ui.main.adapter
+
+import androidx.recyclerview.widget.RecyclerView
+import org.p2p.wallet.databinding.ItemTitleBinding
+import org.p2p.wallet.home.model.HomeElementItem
+
+class TitleViewHolder(
+    private val binding: ItemTitleBinding
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/ui/main/adapter/TitleViewHolder.kt:8]: binding can be called with `Home` in it too
item_home_title for example

File: app/src/main/java/org/p2p/wallet/home/ui/main/adapter/TokenAdapter.kt [hunk 2]
```diff
@@ -37,6 +39,7 @@ class TokenAdapter(
         R.layout.item_token_hidden -> TokenHiddenViewHolder(parent, listener)
         R.layout.item_token_group_button -> TokenButtonViewHolder(parent, listener)
         R.layout.item_banners -> BannersViewHolder(parent, listener)
+        R.layout.item_title -> TitleViewHolder(parent.inflateViewBinding(parent.context, attachToRoot = false))
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/ui/main/adapter/TokenAdapter.kt:42]: inflating can be inside, for the sake of one way creating ViewHolder`s here

File: app/src/main/res/layout/item_token.xml [hunk 3]
```diff
@@ -1,38 +1,34 @@
...
-        android:background="@color/backgroundSecondary"
-        android:foreground="?attr/selectableItemBackground"
-        android:padding="20dp"
-        android:src="@drawable/ic_hide"
-        tools:ignore="ContentDescription" />
+        android:layout_marginEnd="16dp"
+        android:foreground="?selectableItemBackgroundBorderless">
+
+        <ImageView
+            android:id="@+id/imageViewHide"
```
Comment (gslevinkov) [inline app/src/main/res/layout/item_token.xml:20]: imageViewHide -> imageViewHideToken

File: app/src/main/res/layout/item_token.xml [hunk 4]
```diff
@@ -1,38 +1,34 @@
...
-        android:foreground="?attr/selectableItemBackgroundBorderless"
-        android:padding="20dp"
-        android:src="@drawable/ic_send_blue"
-        tools:ignore="ContentDescription" />
-
-    <ImageView
-        android:id="@+id/hideImageView"
-        android:layout_width="64dp"
+    <FrameLayout
+        android:id="@+id/frameLayoutHide"
```
Comment (gslevinkov) [inline app/src/main/res/layout/item_token.xml:12]: frameLayoutHide -> layoutHide

File: app/src/main/res/layout/item_token_hidden.xml [hunk 5]
```diff
@@ -1,38 +1,34 @@
...
-        android:background="@color/backgroundSecondary"
-        android:foreground="?attr/selectableItemBackgroundBorderless"
-        android:padding="20dp"
-        android:src="@drawable/ic_show"
-        tools:ignore="ContentDescription" />
+        android:layout_marginEnd="16dp"
+        android:foreground="?selectableItemBackgroundBorderless">
+
+        <ImageView
+            android:id="@+id/imageViewExpose"
```
Comment (gslevinkov) [inline app/src/main/res/layout/item_token_hidden.xml:20]: imageViewExpose  -> imageViewExposeToken

File: ui-kit/src/main/res/values/typography.xml [hunk 6]
```diff
@@ -14,7 +14,7 @@
     </style>
 
     <style name="UiKit.TextAppearance.Regular.Label2" parent="UiKit.TextAppearance.Regular">
-        <item name="android:textSize">11sp</item>
```
Comment (gslevinkov) [inline ui-kit/src/main/res/values/typography.xml:17]: Label2 is 11sp, it has not been changed at all

**Review by gslevinkov — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/p2p/wallet/home/ui/main/adapter/TokenAdapter.kt (hunk 2 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/home/ui/main/adapter/TokenAdapter.kt:42]: should be - to be in common approach

File: app/src/main/res/layout/item_title.xml [hunk 7]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (kabramovp2p) [inline app/src/main/res/layout/item_title.xml:1]: check please org.p2p.wallet.home.ui.main.adapter.HeaderViewHolder and reuse it if we can
i see SemiBold.Title3 and SemiBold.Text1
but i presume it should be same and we need to force our designers to reuse headers

**Review by gslevinkov — APPROVED** [review]

**Review by kuznetsov — APPROVED** [review]

**Review by kuznetsov — CHANGES_REQUESTED** [review]
Please disable drag gesture on SOL token

**Review by kabramovp2p — APPROVED** [review]

**Pull request merged**